### PR TITLE
Default peer service name to blank as it is required in Thrift Endpoint

### LINKED ISSUE #45: It is possible to report invalid spans (endpoint.serviceName == null)
In zipkin's thrift, Endpoint.serviceName is not an optional field. Conventionally, some set this to empty, but not serializing it is a mistake. I noticed this when using the jersey client, which might hint there's a bug there (as I did specify a serviceName).

This should be enforced in unit tests or similar as otherwise the only way to know is after it reaches zipkin.

### PR DESCRIPTION
**Pull request by yurishkuro**
Fixes #45

### CODE REVIEW

File: jaeger-core/src/main/java/com/uber/jaeger/Span.java [hunk 1]
```diff
@@ -72,7 +72,7 @@ public Tracer getTracer() {
     public Endpoint getPeer() {
         synchronized (this) {
             if (peer == null) {
-                peer = new Endpoint();
+                peer = new Endpoint(0, (short) 0, "");
```
Comment (adriancole) [inline jaeger-core/src/main/java/com/uber/jaeger/Span.java]: might want to return null.. ex null endpoint is ok, it is null serviceName that's the problem. an empty endpoint w/o an ip or serviceName or port, is probably worse than null.

**Comment by adriancole** [comment]
PS verified this fixes it!

**Comment by badiib** [comment]
LGTM pending the null conversation.

File: jaeger-core/src/main/java/com/uber/jaeger/Span.java [hunk 2]
```diff
@@ -21,15 +21,15 @@
  */
 package com.uber.jaeger;
 
+import com.twitter.zipkin.thriftjava.Endpoint;
+import com.uber.jaeger.utils.Utils;
+import io.opentracing.tag.Tags;
```
Comment (yurishkuro) [inline jaeger-core/src/main/java/com/uber/jaeger/Span.java]: intelliJ kept insisting, so I let it

**Comment by yurishkuro** [comment]
OK I figured it out. It was really a bug: reusing getPeer method as both lazy initializer and getter. I split them into plain getter and `getOrMakePeer`, the latter is only used by the methods that actually set the values on the peer, so the defaulting makes a lot more sense.

**Comment by yurishkuro** [comment]
released as 0.8.1

**Comment by adriancole** [comment]
thanks

**Pull request merged**